tests/framework/extensions/provisioning/clusterconfig.go

```diff
@@ -0,0 +1,30 @@
+package provisioning
+
+import (
+	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+	nodepools "github.com/rancher/rancher/tests/framework/extensions/rke1/nodepools"
+)
+
+type ClusterConfig struct {
+	KubernetesVersion    string                                   `json:"kubernetesVersion" yaml:"kubernetesVersion"`
+	CNI                  string                                   `json:"cni" yaml:"cni"`
+	PSACT                string                                   `json:"psact" yaml:"psact"`
+	PNI                  bool                                     `json:"pni" yaml:"pni"`
+	NodesAndRoles        *[]machinepools.NodeRoles                `json:"nodesAndRoles" yaml:"nodesAndRoles" default:"[]"`
+	NodesAndRolesRKE1    *[]nodepools.NodeRoles                   `json:"nodesAndRolesRKE1" yaml:"nodesAndRolesRKE1" default:"[]"`
+	Providers            *[]string                                `json:"providers" yaml:"providers"`
+	NodeProviders        *[]string                                `json:"nodeProviders" yaml:"nodeProviders"`
+	Hardened             bool                                     `json:"hardened" yaml:"hardened"`
+	AddOnConfig          *AddOnConfig                             `json:"addonConfig" yaml:"addonConfig"`
+	AgentEnvVars         *[]management.EnvVar                     `json:"agentEnvVars" yaml:"agentEnvVars"`
```

Comment:
From my testing I think we'll need two flavors of AgentEnvVars, this one should have the RKE1 label, end result should be something like this:
```
AgentEnvVars    []rkev1.EnvVar   `json:"agentEnvVars" yaml:"agentEnvVars" default:"[]"`
AgentEnvVarsRKE1    []management.EnvVar   `json:"AgentEnvVarsRKE1" yaml:"agentEnvVarsRKE1" default:"[]"`
```